agent/config/builder.go

```diff
@@ -1090,6 +1089,7 @@ func (b *builder) build() (rt RuntimeConfig, err error) {
 		ServerMode:                        serverMode,
 		ServerName:                        stringVal(c.ServerName),
 		ServerPort:                        serverPort,
+		ServerRejoinAgeMax:                b.durationValWithDefault("server_rejoin_age_max", c.ServerRejoinAgeMax, 24*7*time.Hour),
```

Comment:
We should validate that it isn't set to anything less than an hour here (because that's how often we write the file out to disk) 💡

Reply:
Something like this? https://github.com/hashicorp/consul/pull/17171/commits/21e26a0e01efffcc5fd6b3e2a660dc2a63557558

Reply:
Nice! Although, maybe this check should be performed in [`builder.validate`](https://github.com/hashicorp/consul/blob/d0beddd9d141868d37b9415a28d78b94fd3d0753/agent/config/builder.go#L1196) - sorry that's my bad, should've been more specific than saying "here" 😅 

We should also add a test case to [`TestLoad_IntegrationWithFlags`](https://github.com/hashicorp/consul/blob/boxofrad/resource-documentation/agent/config/runtime_test.go#L71).

Reply:
Wondering if we want to add a cli flag for this? I'm not sure what the criteria is vs just having a config option only.

Reply:
I think we can probably skip the CLI flag for this, as users will very infrequently need to change it.